client/allocdir/alloc_dir.go

```diff
@@ -126,18 +126,24 @@ func (d *AllocDir) Embed(task string, dirs map[string]string) error {
 			continue
 		}
 
-		// Enumerate the files in source.
-		entries, err := ioutil.ReadDir(source)
-		if err != nil {
-			return fmt.Errorf("Couldn't read directory %v: %v", source, err)
+		// Check if destination directory exists. This can happen if restarting
+		// a failed task.
+		destDir := filepath.Join(taskdir, dest)
+		if _, err := os.Stat(destDir); err == nil {
```

Comment:
What happens if nomad failed halfway through copying the files into place? For the sake of idempotency it might be best to run through the rest of the procedure here.